Update/settings page ui

leonardola opened the pull request:
Related to #5372
Fix #5161

#### Changes proposed in this Pull Request
This PR updates the express checkouts ui so it follows the new designs from figma (C1r0Vzxh1mhbDJJnbjHp5q-fi-1437%3A71323)

#### Testing instructions

- Go to wp-admin > Payments > Settings >  Express Checkout and make sure the ui looks the same as the design in the figma file
- Change the setting and make sure they reflect in the product, cart and checkout pages. Also test the blocks pages.
-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Commit:
removed backend settings for woopay

Commit:
Now it is possible to enable WooPay and Google/Apple buttons separately

Commit:
now the settings are shared between WooPay and Google/Apple pay buttons

Commit:
moved the Show express checkouts on to the activation section

Commit:
fixed some tests

Commit:
fixed final js tests

Commit:
fixed php tests

Commit:
removed empty test file

Commit:
added changelog

Commit:
added notice for shared settings

Commit:
fixed show on pages title

Commit:
fixed show on pages title

Commit:
added WooPay preview, Fixed style to reflect design

Commit:
fixed Apple/Google Pay icons in the settings page

Commit:
moved the general settings to a separate file so we can reuse it for both express checkouts

Commit:
fixed WooPay button style

Commit:
fixed Apple/Google pay icons in the general settings page

Commit:
fixed copy

Commit:
fixed button text

Commit:
fixed missing button text

Commit:
fixed some tests

Commit:
Fixed tests

Commit:
fixed bug that would show WooPay button even when it was deactivated for that page on blocks checkout/cart

Commit:
added changelog

Commit:
Merge branch 'develop' into update/settings-page

# Conflicts:
#	client/settings/express-checkout-settings/payment-request-settings.js
#	client/settings/express-checkout-settings/platform-checkout-settings.js
#	client/settings/express-checkout-settings/test/payment-request-settings.test.js
#	includes/class-wc-payments-platform-checkout-button-handler.php

Commit:
Merge branch 'develop' into update/settings-page

malithsen commented on client/checkout/platform-checkout/express-button/woopay-express-checkout-payment-method.js:
```diff
@@ -34,7 +34,13 @@ const wooPayExpressCheckoutPaymentMethod = () => ( {
 		/>
 	),
-	canMakePayment: () => true,
+	canMakePayment: () => {
```
nit: a bit more concise version
```suggestion
	canMakePayment: () => typeof wcpayConfig !== 'undefined'
```

malithsen commented on client/settings/deposits/index.js:
```diff
@@ -238,10 +238,7 @@ const Deposits = () => {
-							'Bank account information',
-							'woocommerce-payments'
-						) }
+						{ __( 'Deposit bank account', 'woocommerce-payments' ) }
```
Just wanted to verify this text change is intentional, as it doesn't seem to be related to the rest of the changes here.

leonardola replied:
Yep. It's in the Figma files. So I guess it would be ok to also update this.

malithsen reviewed (commented):
@leonardola Thanks for the work here. This works great. I noticed a couple of mismatches with the Figma screens and wanted to highlight them in case we want to address them here.

* When "Only Icon" is selected, the top and bottom padding are too low compared to what's in the Figma screens. It shows up correctly on the actual pages.

![Screen Shot 2023-01-31 at 3 00 40 PM](https://user-images.githubusercontent.com/6216000/215885622-a016d034-a1ea-4745-b4a5-17413f676d88.png)

* Button hover state is not respected. This applies to when the WooPay button is displayed on actual pages as well.

https://user-images.githubusercontent.com/6216000/215886266-57444086-2d79-4785-84b5-27e0cccb54cc.mov

* Not sure about this one. Should we collapse the buttons on smaller screens? (Vertical Layout in Figma)



![Screen Shot 2023-01-31 at 3 24 10 PM](https://user-images.githubusercontent.com/6216000/215886813-c36b024d-3098-4c25-8dec-345d397d4a7b.png)

Commit:
removed unnecessary lambda function

Commit:
fixed button height when using the icon only layout

Commit:
fixed WooPay button icon for lighter themes, button background color, button hover background color, WooPay logo alignment,button font sizes, WooPay copy, button preview size on mobile, button label alignment, preview background for light themes

Commit:
fixed WooPay copy links

Commit:
Merge branch 'develop' into update/settings-page

leonardola commented:
@malithsen I've fixed the problems you pointed. Also I noticed that more stuff was added to the figma file so I've also updated the other things too. I only left number 3 out because apple requires that Apple Pay logo "box" to not be shared with other elements and number 5 and number 6 because those are styles inherited from Wordpress itself. So I don't think it's correct to update them so I've asked design about this on slack( p1676323567798829-slack-C01TNBRTMEU ) and if needed we'll create another PR to fix them so this one can get merged sooner. 

Could you please take another look at this PR? If you can't review it soon please ping @a-danae to do so as we need this merged for 5.5. 

Thanks in advance.

malithsen commented on client/checkout/platform-checkout/express-button/woopay-express-checkout-payment-method.js:
```diff
@@ -34,13 +34,7 @@ const wooPayExpressCheckoutPaymentMethod = () => ( {
-
-		return true;
-	},
+	canMakePayment: 'undefined' === typeof wcpayConfig,
```
Shouldn't this be `!==`?

malithsen reviewed (commented):
Thanks for making those updates @leonardola! Looks great to me. One question on the updated operator in https://github.com/Automattic/woocommerce-payments/pull/5490/commits/8a3a09a4cf07974e4b9128bc593bc79131fd65a0, after adressing that, we should be good to merge.

Commit:
fixed the check to allow making payments

Commit:
Merge branch 'develop' into update/settings-page

malithsen commented on client/checkout/platform-checkout/express-button/woopay-express-checkout-payment-method.js:
```diff
@@ -34,7 +34,7 @@ const wooPayExpressCheckoutPaymentMethod = () => ( {
 		/>
 	),
-	canMakePayment: () => true,
+	canMakePayment: 'undefined' !== typeof wcpayConfig,
```
```suggestion
    canMakePayment: () => 'undefined' !== typeof wcpayConfig,
```

apparently, `canMakePayment` should be a function, otherwise WooPay express button in Blocks checkout doesn't show up

malithsen reviewed (changes requested):
Ran into an issue while testing something else. It can be fixed easily. I am less confident about resolving the merge conflicts, so didn't push the fixes myself.

Commit:
Merge branch 'develop' into update/settings-page

# Conflicts:
#	client/settings/express-checkout/index.js
#	client/utils/checkout.js
#	includes/class-wc-payments-upe-checkout.php

Commit:
added function to validate if the WooPay express checkout button should actually open WooPay when clicked. Fixed cursor type on hover

leonardola commented:
@malithsen I've fixed the problem and merged develop into this branch. Could you take a final look and approve it if it looks good? Please go ahead and merge it when approved if no conflicts show up till then. Thanks.

malithsen reviewed (approved):
Thanks for all your work here @leonardola This looks good to me now
